app/utils/time_tools.js

```diff
@@ -0,0 +1,23 @@
+export function getFormattedDateTimeString(date, time, timezone) {
+    validateDateTimeFormat(date, time, timezone);
+    return `${date}T${time}${timezone}`;
+}
+
+function validateDateTimeFormat(date, time, timezone) {
+    const dateFormatRegex = /^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/;
+    const timeFormatRegex = /^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$/;
+    const timezoneFormatRegex = /^[+-]((0[0-9]|1[0-1]):[0-5][0-9])|12:00$/;
```

Comment:
not sure if it is relevant, but with the Regex, even a timezone such as _+03:34_ would work (and get handled correctly) even tho it clearly isn't a timezone.